### PR TITLE
Issues/1190

### LINKED ISSUE #1190: CLI Improvement: Ability to filter donors by form
## Issue Overview

In your CLI integration, it would be nice to be able to limit the donor list to those who have donated via a specific form.

## Expected Behavior

It would be nice to be able to specify a form ID and have the results limited to donors who contributed via that form.

## Current Behavior

The donors command currently lists all donors, regardless of which form they used. There is no way to filter those results.

## Possible Solution

I would suggest adding a --form-id parameter to the donors command that would filter the results, example:

`wp give donors --form-id=7`

### PR DESCRIPTION
**Pull request by ravinderk**
## Description
This pr will fix #1190 

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

**Comment by ravinderk** [comment]
@DevinWalker Please review this commit https://github.com/WordImpress/Give/pull/1202/commits/a17acc7ea40273ab04f7f350dea31d9d5f899df6.

I changed `[give_forms](https://github.com/ravinderk/Give/blob/a17acc7ea40273ab04f7f350dea31d9d5f899df6/includes/payments/class-payments-query.php#L515#L527)` functions. Previously it was getting donation IDs from logs and now I am adding meta query when user query for specific form related donations.

#### Suggestions
I think logs are not static and we should provide a tool:
1. To flush them automatically
2. Give the option to enable/disable logs

#### Question
Currently, user can fetch a list of donors for s specific form of all completed donations.
Do we have to create extra params to `donors` subcommand which filter a list of donors on basis of donation status?

What do you think?

**Comment by DevinWalker** [comment]
@ravinderk can you resolve the conflict here?

**Comment by ravinderk** [comment]
@DevinWalker conflict resolved.

**Comment by DevinWalker** [comment]
@ravinderk thanks!

File: includes/payments/class-payments-query.php [hunk 1]
```diff
@@ -512,43 +512,19 @@ public function give_forms() {
 			return;
 		}
 
-		global $give_logs;
-
-		$args = array(
```
Comment (DevinWalker) [inline includes/payments/class-payments-query.php]: I see you're cleaning up the code here, which is nice. Have you tested this throughout the ensure it works as expected?

Comment (ravinderk) [inline includes/payments/class-payments-query.php]: @DevinWalker I did not find the use of this code inside core but it is working fine with give-cli.

File: includes/payments/class-payments-query.php [hunk 2]
```diff
@@ -512,43 +512,19 @@ public function give_forms() {
...
-		);
+		$compare = '=';
 
 		if ( is_array( $this->args['give_forms'] ) ) {
-			unset( $args['post_parent'] );
-			$args['post_parent__in'] = $this->args['give_forms'];
+			$compare = 'IN';
 		}
 
-		$sales = $give_logs->get_connected_logs( $args );
```
Comment (ravinderk) [inline includes/payments/class-payments-query.php]: Yes

**Pull request merged**